File: textbook-codes/Kap_03_Curvefit.jl
#-- Kap_03_Curvefit.jl
using LsqFit, Plots
#--
x = 0.0:0.15:2*pi; y = 3*sin.(4*x)+0.5*rand(length(x));
scatter(x, y, label="Wertepaare")
model(t, p) = p[1]*sin.(p[2]*t)
p0 = [2.0, 3.5]
fit = curve_fit(model, x, y, p0)
pp = fit.param
xx = 0:0.01:2*pi; yy = model(xx,pp)
p=plot!(xx,yy, label="Curve Fit Julia",xlabel = "x",ylabel="y",
               linewidth=2,dpi=300,show=true)

File: textbook-codes/Kap_03_Curvefit.m
%-- Kap_03_Curvefit.m
x = 0.0:0.15:2*pi; y = 3*sin(4*x)+0.5*rand(1,length(x));
f = fittype('a*sin(b*x)'); 
pp = fit(x',y',f,'StartPoint',[2 3.5])
xx = 0:0.01:2*pi;
yy = pp.a*sin(pp.b*xx);
plot(x,y,'o',xx,yy,'linewidth',2)
title('Curfe Fit MATLAB'); xlabel('x'); ylabel('y');
